orttraining/orttraining/test/training_ops/cuda/optimizer/adamw_test.cc

```diff
@@ -0,0 +1,928 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+
+#include "test/providers/provider_test_utils.h"
+#include "orttraining/test/training_ops/cuda/optimizer/common.h"
+
+namespace onnxruntime {
+namespace test {
+namespace optimizer {
+
+#if USE_CUDA
+
+namespace {
+
+void TorchAdamWSingleWeightTestLoop10Steps(bool use_baseline_for_each_iteration) {
+  size_t total_step = 10;
+  float lr = 1e-03f;
+
+  std::pair<float, float> weight_tolerance{1e-4f, 1e-5f};  // rtol, atol
+  std::pair<float, float> momentum1_tolerance{1e-3f, 1e-6f};
+  std::pair<float, float> momentum2_tolerance{1e-2f, 1e-7f};
+
+  if (!use_baseline_for_each_iteration) {
+    // Loose the tolerance as all states are maintained (without reloading from baseline) across different steps.
+    momentum2_tolerance.first = 1e-3f;
+    momentum2_tolerance.second = 1e-6f;
+  }
+
+  // Test data generated by onnxruntime/test/testdata/optimizers/adamw_test_data_generator.py
+
+  // 11 steps of weight values before applying optimization.
+  std::unordered_map<std::string, std::vector<std::vector<float>>> named_weights{
+      {
+          "fc1.weight",
+          {
+              {-0.18330415f, 0.6739549f, 0.3117089f, 0.42830977f, -0.39579117f, 0.07424858f},
```

Comment:
maybe these values should be saved/read from files instead of relying on copying from std::out ?

Reply:
done.

Reply:
saved in json and loaded in tests.